vault-plugins/flant_iam/model/role_test.go

```diff
@@ -10,3 +63,36 @@ func Test_RoleDbSchema(t *testing.T) {
 		t.Fatalf("role schema is invalid: %v", err)
 	}
 }
+
+func createRoles(t *testing.T, repo *RoleRepository, roles ...Role) {
+	for _, role := range roles {
+		tmp := role
+		err := repo.Create(&tmp)
+		dieOnErr(t, err)
+	}
+}
+
+func Test_Role_findDirectIncludingRoles(t *testing.T) {
+	schema := RoleSchema()
+	store, err := io.NewMemoryStore(schema, nil)
+	dieOnErr(t, err)
+	tx := store.Txn(true)
+	defer tx.Abort()
+	repoRole := NewRoleRepository(tx)
+	createRoles(t, repoRole, []Role{role1, role2, role3, role4, role5}...)
+	roles, err := repoRole.findDirectIncludingRoles(roleName1)
+	dieOnErr(t, err)
+	checkDeepEqual(t, map[string]struct{}{roleName3: {}, roleName4: {}}, roles)
+}
+
+func Test_Role_FindAllIncludingRoles(t *testing.T) {
+	schema := RoleSchema()
+	store, err := io.NewMemoryStore(schema, nil)
+	dieOnErr(t, err)
+	tx := store.Txn(true)
+	repoRole := NewRoleRepository(tx)
+	createRoles(t, repoRole, []Role{role1, role2, role3, role4, role5}...)
+	roles, err := repoRole.FindAllIncludingRoles(roleName1)
+	dieOnErr(t, err)
+	checkDeepEqual(t, map[string]struct{}{roleName3: {}, roleName4: {}, roleName5: {}}, roles)
+}
```

Comment:
In tests, I suggest leaving blank lines betwee setup, action, and assertion, so one could easier navigate the three parts of a test. I'd prefer to see the structure.

```go
/*
func test() {
  setup

  action

  assertion
}
*/
func Test_Role_FindAllIncludingRoles(t *testing.T) {
	schema := RoleSchema()
	store, err := io.NewMemoryStore(schema, nil)
	dieOnErr(t, err)
	tx := store.Txn(true)
	repoRole := NewRoleRepository(tx)

	createRoles(t, repoRole, []Role{role1, role2, role3, role4, role5}...)
	roles, err := repoRole.FindAllIncludingRoles(roleName1)

	dieOnErr(t, err)
	checkDeepEqual(t, map[string]struct{}{roleName3: {}, roleName4: {}, roleName5: {}}, roles)
}
```

Reply:
fixed

Reply:
fixed